### PR TITLE
[bitnami/mongodb] clusterDomain values forces to use only FQDN

### LINKED ISSUE #2312: [bitnami/mongodb] clusterDomain values forces to use only FQDN
**Which chart**:
mongodb (7.10.10)

**Is your feature request related to a problem? Please describe.**
There is a value: `clusterDomain` which is used in:
https://github.com/bitnami/charts/blob/master/bitnami/mongodb/templates/statefulset-arbiter-rs.yaml#L111
https://github.com/bitnami/charts/blob/master/bitnami/mongodb/templates/statefulset-primary-rs.yaml#L126
https://github.com/bitnami/charts/blob/master/bitnami/mongodb/templates/statefulset-secondary-rs.yaml#L129

The implementation always uses the FQDN to resolve the DNS. i.e:
- `mongo-headless.svc.cluster.local`
or
- `mongo-headless.svc.[CUSTOM_DOMAIN]` [THIS IS MY CASE]

This makes to not use the default kubernetes options for resolving dns:
1. `svc_name` 
2. `svc_name.namespace` 
3. `svc_name.namespace.svc.cluster_domain` [THIS IS THE ONLY POSSIBILITY]

This makes the chart to loose versatility. 

**Describe the solution you'd like**
In my personal case, due to cluster setup, it would be handier if this chart offers options `1.` and `2.`
So that I could not be forced to set a value for `clusterDomain` always. Option `1.` (resolution by service name) would be enough.

**Describe alternatives you've considered**
Instead of adding a default value in [clusterDomain value](https://github.com/bitnami/charts/blob/master/bitnami/mongodb/values.yaml#L122), just leave it empty.
Then in the template, something like:
```
{{- if .Values.replicaSet.clusterDomain }}
  - name: MONGODB_ADVERTISED_HOSTNAME
    value: "$(MONGODB_POD_NAME).{{ template "mongodb.fullname" . }}-headless.{{ .Release.Namespace }}.svc.{{ .Values.clusterDomain }}"
{{- else }}
  - name: MONGODB_ADVERTISED_HOSTNAME
    value: "$(MONGODB_POD_NAME).{{ template "mongodb.fullname" . }}-headless.{{ .Release.Namespace }}"
{{- end }}
```

### PR DESCRIPTION
**Pull request by antonioberben**
**Description of the change**

So far, user of chart was forced to use only FQDN since the default value of `clusterDomain` was `cluster.local`.  This is either:
- `[svc_name].[namespace].svc.[cluster.local]`
or:
- `[svc_name].[namespace].svc.[clusterDomain]`

With the change it can use the namespaced way:
-  `[svc_name].[namespace]`

**Benefits**

This is closer to what kubernetes offers, the three ways to resolve dns:
- `[svc_name].[namespace].svc.[clusterDomain]`
- `[svc_name].[namespace]`
- `[svc_name]`

For our concrete case, some providers set a specific clusterDomain to the cluster.  i.e. if you have two clusters the clusterDomain are `cluster1` and `cluster2`

Therefore, once intalling this chart in `cluster1` without this change, I am forced to set the value as parameter `--set clusterDomain=cluster1` and `--set clusterDomain=cluster2`

That is one of the reasons kubernetes offers the three ways of resolving domains. So that installations can be more unified.

With this PR change, those two installations explain before will be same (no need of setting the clusterDomain value)

**Possible drawbacks**

Misfunctionality in the way mongodb works internally

**Applicable issues**

  - fixes #2312 

**Additional information**

N/A

**Checklist** 
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [x] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

**Comment by antonioberben** [comment]
Hi @dani8art , I'm sorry, but I don't understand what you mean with chaging the service names.
Do you mean in [here](https://github.com/bitnami/charts/pull/2340/files#diff-15048183f6906cede0ba87b0149d8730L111)? 
Cheers!

**Comment by dani8art** [comment]
Yes sorry, it is right!

File: bitnami/mongodb/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: mongodb
-version: 7.10.12
+version: 7.10.13
```
Comment (dani8art) [inline bitnami/mongodb/Chart.yaml:3]: I would bump a minor version because we are modifying the behavior of a value.

Comment (antonioberben) [reply bitnami/mongodb/Chart.yaml]: done with last version of master

**Comment by dani8art** [comment]
Hi @antonioberben 

In order to update the image tags to point to latest releases, our CI/CD needs to modify you PR (`bitnami-bot` will update `values.yaml` and `values-production.yaml` for you). To do so, it's necessary you allow "edits from maintainers". Find more information in the link below:
- https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork

**Pull request merged**